src/Airship/Config.hs

```diff
@@ -0,0 +1,30 @@
+module Airship.Config
+    ( AirshipConfig
+    , includeTraceHeader
+    , defaultAirshipConfig
+    ) where
+
+import           Lens.Micro (Lens', lens)
+
+-- | An opaque data type encapsulating all Airship-specific configuration options.
+--
+-- We use lenses to modify 'AirshipConfig' values -- though Airship only depends on the
+-- microlens library, its lenses are compatible with Control.Lens.
+data AirshipConfig = AirshipConfig
+    { _includeTraceHeader :: Bool
```

Comment:
Sorry to intrude. One minor suggestion/idea - we tend to use datatypes instead of `Bool` where possible, which has many benefits like documentation and stronger types. It might be overkill for this field, but just thought I would mention it. :)